### DISCUSSION TITLE
Trace to logs: Tags in traces

### DISCUSSION THREAD

**Discussion by sandhraprakash** [open-telemetry/opentelemetry-java-instrumentation#5076] [Q&A]
I am using opentelemetry java instrumentation to auto instrument tracing in my application. and tempo is what I use for backend. and Grafana for visualization. 
In Grafana tempo, we can view logs corresponding to a trace. https://grafana.com/docs/grafana/latest/datasources/tempo/#trace-to-logs. In order for this to work, the label in log should be available as tag in trace. 
So my question is, Is there a way to add tags to traces using auto instrumentation?

**Comment by trask** [comment]
hi @sandhraprakash! is this what you're looking for? https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/logger-mdc-instrumentation.md#logger-mdc-auto-instrumentation

**Reply by sandhraprakash** [reply]
No, This is how I can write traceId and span Id into my logs, What I am looking for is to add Tags (Key Value pair) under the span itself. 
If it was manually instrumented, the tag would be added using AddTag api or something close for the Span.
Hope the image gives a better Idea,
![image](https://user-images.githubusercontent.com/34446301/149089350-d7e70f40-915b-49a7-a297-dfa623afd4f2.png)

**Reply by trask** [reply]
ah, you can add span tags ("attributes") using `Span.current().setAttribute("key", "value")`

**Reply by sandhraprakash** [reply]
I will have to add a spanProcessor and set the attribute in onStart? The spanProcessor then should be added to SDKTraceProvider? I am asking this because its auto instrumentation, we havent added any open telemetry dependency to code. So do I have to do a similar setup like that in manual instrumentation?

**Reply by trask** [reply]
hi @sandhraprakash, you can add the `opentelemetry-api` artifact to your application, and call `Span.current().setAttribute("key", "value")`. this will get the current span (from the javaagent) and set an attribute on it
